src/reformat_libyuv.c

```diff
@@ -707,166 +704,113 @@ static avifResult avifImageDownshiftTo8bpcYUVAndConvertToRGBLibYUV(const avifIma
+                          image8->yuvRowBytes[plane],
                           scale,
                           planeWidth,
                           planeHeight);
     }
-    result = apply8bpcLibYUVConversionFunction(&image8, rgb, matrixYUV, matrixYVU, &yuv400ToRgbMatrix, &yuvToRgbMatrixFilter, &yuvToRgbMatrix);
-    avifImageFreePlanes(&image8, AVIF_PLANES_YUV);
-    return result;
+    return AVIF_RESULT_OK;
 }
 
-avifResult avifImageYUVToRGBLibYUVHighBitDepth(const avifImage * image,
-                                               avifRGBImage * rgb,
-                                               const struct YuvConstants * matrixYUV,
-                                               const struct YuvConstants * matrixYVU)
+avifResult avifImageYUVToRGBLibYUV(const avifImage * image, avifRGBImage * rgb)
 {
-    // See if the current settings can be accomplished with libyuv, and use it (if possible).
-
-    assert(((image->depth == 10) || (image->depth == 12)) && (rgb->depth == 8));
-    int depthIndex = (image->depth == 10) ? 0 : 1;
-
-    // libavif uses byte-order when describing pixel formats, such that the R in RGBA is the lowest address,
-    // similar to PNG. libyuv orders in word-order, so libavif's RGBA would be referred to in libyuv as ABGR.
-    // In addition, swapping U and V in any of these calls, along with using the Yvu matrix instead of Yuv matrix,
-    // swaps B and R in these orderings as well. This table summarizes the lookup tables that follow:
-    //
-    // libavif format            libyuv Func      UV matrix (and UV argument ordering)
-    // --------------------      -------------    ------------------------------------
-    // AVIF_RGB_FORMAT_RGB       n/a              n/a
-    // AVIF_RGB_FORMAT_RGBA      *ToARGBMatrix    matrixYVU
-    // AVIF_RGB_FORMAT_ARGB      n/a              n/a
-    // AVIF_RGB_FORMAT_BGR       n/a              n/a
-    // AVIF_RGB_FORMAT_BGRA      *ToARGBMatrix    matrixYUV
-    // AVIF_RGB_FORMAT_ABGR      n/a              n/a
-    // AVIF_RGB_FORMAT_RGB_565   n/a              n/a
+    if (rgb->depth != 8 || (image->depth != 8 && image->depth != 10 && image->depth != 12)) {
+        return AVIF_RESULT_NOT_IMPLEMENTED;
+    }
+    // Find the correct libyuv YuvConstants, based on range and CP/MC
+    const struct YuvConstants * matrixYUV = NULL;
+    const struct YuvConstants * matrixYVU = NULL;
+    getLibYUVConstants(image, &matrixYUV, &matrixYVU);
+    if (!matrixYVU) {
+        // No YuvConstants exist for the current image; use the built-in YUV conversion
+        return AVIF_RESULT_NOT_IMPLEMENTED;
+    }
 
+    LibyuvConversionFunction lcf;
+    if (!getLibYUVConversionFunction(image->yuvFormat, image->depth, rgb, &lcf)) {
+        return AVIF_RESULT_NOT_IMPLEMENTED;
+    }
     avifBool isYVU = lutIsYVU[rgb->format];
     const struct YuvConstants * matrix = isYVU ? matrixYVU : matrixYUV;
-    if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400) {
```

Comment:
The special handling of `AVIF_PIXEL_FORMAT_YUV400` in the high bit depth case is also missing in the new code. The new code relies on the lookup tables having `NULL` entries for `AVIF_PIXEL_FORMAT_YUV400`, so it is still correct.

Reply:
yes, that table does not exist as of now. so i think this is fine.